packer/plugin.go

```diff
@@ -152,8 +171,17 @@ func (c *PluginConfig) DiscoverMultiPlugin(pluginName, pluginPath string) error
 		if c.Builders.Has(key) {
 			continue
 		}
+		registered = true
+
 		c.Builders.Set(key, func() (packersdk.Builder, error) {
-			return c.Client(pluginPath, "start", "builder", builderName).Builder()
+			args := []string{"start", "builder"}
+
+			if PackerUseProto {
```

Comment:
Should this be canProto? It looks like PackerUseProto is always true, which means we will always send --protobuf.

Reply:
Good question; nope :p

We use `PackerUseProto` as it's global, since this is a function that is lazily evaluated, we cannot access `canProto` as the value is lost once we leave this function (it's all local).
However if the value of `PackerUseProto` is changed to `false` while discovering plugins, when this is executed, it will use the last updated value for this, and correctly pick between protobuf/gob.